indigo-plugin/indigo-plugin/src/indigoplugin/generators/FontGen.scala

```diff
@@ -0,0 +1,302 @@
+
+    // Write out font image
+    val outImageFileName = s"$moduleName.png"
+
+    helper.drawFontSheet((imageOut / outImageFileName).toIO, charDetails, sheetWidth, sheetHeight, fontOptions)
+
+    Seq(file)
+  }
+
+  def sanitiseName(name: String, ext: String): String = {
+    val noExt = if (ext.nonEmpty && name.endsWith(ext)) name.dropRight(ext.length) else name
+    noExt.replaceAll("[^A-Za-z0-9]", "-").split("-").map(_.capitalize).mkString
+  }
+
+  // TODO: Does nothing
+  def layout(unplacedChars: List[CharDetail], lineSpacing: Int, maxCharsPerLine: Int): List[CharDetail] = {
+    @tailrec
+    def rec(
+        remaining: List[CharDetail],
+        lineCount: Int,
+        charCount: Int,
+        nextX: Int,
+        acc: List[CharDetail]
+    ): List[CharDetail] =
+      remaining match {
+        case Nil =>
+          acc.reverse
+
+        case c :: cs if charCount == maxCharsPerLine =>
+          rec(c :: cs, lineCount + 1, 0, 0, acc)
+
+        case c :: cs =>
+          val x    = nextX
+          val y    = lineCount * lineSpacing
+          val newC = c.copy(x = x, y = y)
+
+          rec(cs, lineCount, charCount + 1, nextX + c.width, newC :: acc)
+      }
+
+    rec(unplacedChars, 0, 0, 0, Nil)
+  }
+
+  def findBounds(charDetails: List[CharDetail]): (Int, Int) =
+    charDetails.foldLeft((0, 0)) { case ((w, h), c) =>
+      (
+        if (c.x + c.width > w) c.x + c.width else w,
+        if (c.y + c.height > h) c.y + c.height else h
+      )
+    }
+
+  def genFontInfo(
+      moduleName: String,
+      fullyQualifiedPackage: String,
+      name: String,
+      sheetWidth: Int,
+      sheetHeight: Int,
+      default: CharDetail,
+      chars: List[CharDetail]
+  ): String = {
+
+    val charString = chars
+      .map(cd => "          " + toFontChar(cd) + ",")
+      .mkString("\n")
+      .dropRight(1) // Drops the last ','
+
+    val dx = default.x.toString
+    val dy = default.y.toString
+    val dw = default.width.toString
+    val dh = default.height.toString
+
+    s"""package $fullyQualifiedPackage
+    |
+    |import indigo.*
+    |
+    |// DO NOT EDIT: Generated by Indigo.
+    |object $moduleName {
+    |
+    |  val fontKey: FontKey = FontKey("$name")
+    |
+    |  val fontInfo: FontInfo =
+    |    FontInfo(
+    |      fontKey,
+    |      $sheetWidth,
+    |      $sheetHeight,
+    |      FontChar("${default.char.toString()}", $dx, $dy, $dw, $dh)
+    |    ).isCaseSensitive
+    |      .addChars(
+    |        Batch(
+    |$charString
+    |        )
+    |      )
+    |
+    |}
+    |""".stripMargin
+  }
+
+  def toFontChar(
+      charDetail: CharDetail
+  ): String = {
+    val c = escapeChar(charDetail.char)
+    val x = charDetail.x.toString()
+    val y = charDetail.y.toString()
+    val w = charDetail.width.toString()
+    val h = charDetail.height.toString()
+
+    s"""FontChar("$c", $x, $y, $w, $h)"""
+  }
+
+  def filterUnsupportedChars(c: Char, code: Int): Boolean =
```

Comment:
If my understanding is good, these characters are not included in the font sheet.
Maybe all the control characters, so codes from 0 to 31 should be excluded?
Or is there a use case for some of them?
After some googling, it seems that some fonts have such kind of support but it seems to be very specific : 
https://graphicdesign.stackexchange.com/questions/57352/font-for-representing-unicode-non-printable-characters

Reply:
Yeah I don't know, honestly.

I'm at the point with this where this improvement adds a lot of value - imperfect as it is - and I think I'm just going to unleash it into the wild and wait for the issues to emerge that help shape it's future direction.